vertx-web-client/src/test/java/io/vertx/ext/web/client/it/UriTemplateITest.java

```diff
@@ -0,0 +1,90 @@
+package io.vertx.ext.web.client.it;
+
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static io.vertx.core.Vertx.vertx;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(VertxUnitRunner.class)
+public class UriTemplateITest {
+
+  private HttpClient client = vertx().createHttpClient();
+
+  private Vertx vertx;
+
+  @Before
+  public void deploy(TestContext context) {
+    vertx = Vertx.vertx();
+    Async async = context.async();
+    vertx.deployVerticle(UriTemplateVerticle.class.getName());
```

Comment:
no need for async there, simply use

vertx.deployVerticle(UriTemplateVerticle.class.getName(), context.asyncAssertSuccess())

it creates an async that is completed when the verticle is deployed

Reply:
I tried but that way gave me a TimeoutException and the tests weren't executed, the execution state keeps running on the onComplete from CompositeFuture.all in the UriTemplateVerticle...
For that reason, I used Async, and then everything works fine.

Reply:
ok, I'm not sure why, it should work the way I described

Reply:
I'll take a look, maybe changing a bit the logic about the use of CompositeFuture.all() in my verticle class I can use directly vertx.deployVerticle(UriTemplateVerticle.class.getName(), context.asyncAssertSuccess()) without Asyn object...

Reply:
I did some code refactoring in the verticle and now works well with your suggestion, thanks @vietj .